core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala

```diff
@@ -47,6 +51,23 @@ private[spark] class KVStoreProtobufSerializer extends KVStoreScalaSerializer {
       ApplicationInfoWrapperSerializer.deserialize(data).asInstanceOf[T]
     case _ if classOf[RDDStorageInfoWrapper].isAssignableFrom(klass) =>
       RDDStorageInfoWrapperSerializer.deserialize(data).asInstanceOf[T]
+    case other if KVStoreProtobufSerializer.isSQLExecutionUIData(other.getName) =>
+      KVStoreProtobufSerializer.uiDataSerializer.deserialize(data).asInstanceOf[T]
     case other => super.deserialize(data, klass)
   }
 }
+
+object KVStoreProtobufSerializer {
+  private lazy val uiDataSerializer: ExtendedSerializer = {
```

Comment:
It seems that the initialization way is not elegant. Let me see if there is a better way